ibis/backends/snowflake/__init__.py

```diff
@@ -128,6 +128,21 @@ class Backend(AlchemyCrossSchemaBackend, CanCreateDatabase, AlchemyCanCreateSche
 
     _latest_udf_python_version = (3, 10)
 
+    def _fqdn(self, name, database=None):
+        if database is None:
+            return self._quote(name)
+        pattern = r'(?<!\.)\.(?![^.]*")'
```

Comment:
We can avoid this regex hacking by using `sqlglot.parse_one`. I'll push up the changes.